Tasks/ServiceFabricDeploy/Create-DiffPackage.psm1

```diff
@@ -0,0 +1,149 @@
+        {
+            Remove-Item -Path $diffPackagePath -Recurse -Force
+        }
+        $diffPackagePath = New-Item -ItemType Directory -Path $diffPackagePath -Force
+        $diffPkgAppManifestPath = Join-Path $diffPackagePath $appManifestName
+
+        $clusterAppTypeVersion = $app.ApplicationTypeVersion
+
+        # If the ApplicationTypeVersion of the Application is not upgraded, no diff package is made because there is no need
+        if ($clusterAppTypeVersion -eq $localAppTypeVersion)
+        {
+            Write-Host (Get-VstsLocString -Key DIFFPKG_ApplicationIsNotChanged -ArgumentList @($ApplicationName, $clusterAppTypeVersion, $ConnectedServiceEndpoint.Url))
+            Return
+        }
+
+        # Get the service types from the cluster
+        $serviceTypes = Get-ServiceFabricServiceType -ApplicationTypeName $applicationTypeName -ApplicationTypeVersion $clusterAppTypeVersion
+        # Pack the service manifest names into an array
+        $clusterServiceManifestNames = $serviceTypes.ServiceManifestName
+
+        # If $clusterServiceManifestNames is null, it means no service types are registered. Diff Package is equal to Full Package. Use Full Package to do deployment
+        if (!$clusterServiceManifestNames)
+        {
+            Write-Host (Get-VstsLocString -Key DIFFPKG_NoServicesRunning -ArgumentList @($ApplicationName, $ConnectedServiceEndpoint.Url))
+            Return
+        }
+
+        Write-Host (Get-VstsLocString -Key DIFFPKG_CopyingToDiffPackge -ArgumentList @($localAppManifestPath, $diffPkgAppManifestPath))
+        Copy-Item $localAppManifestPath $diffPkgAppManifestPath -Force
+
+        # Get the service manifests from the cluster
+        $clusterServiceManifestByName = @{}
+        foreach ($clusterServiceManifestName in $clusterServiceManifestNames)
+        {
+            $clusterServiceManifestContent = Get-ServiceFabricServiceManifest -ApplicationTypeName $applicationTypeName -ApplicationTypeVersion $clusterAppTypeVersion -ServiceManifestName $clusterServiceManifestName
+            $clusterServiceManifestByName[$clusterServiceManifestName] = [XML]$clusterServiceManifestContent
+        }
+
+        foreach ($serviceManifestImport in $localAppManifestXml.ApplicationManifest.ServiceManifestImport)
+        {
+            # Open the service manifest associated with the current ServiceManifestImport element of the local ApplicationManifest
+            $serviceManifestName = "ServiceManifest.xml"
+            $localServicePkgPath = Join-Path $ApplicationPackagePath $serviceManifestImport.ServiceManifestRef.ServiceManifestName
+            $localServiceManifestPath = [System.IO.Path]::Combine($localServicePkgPath, $serviceManifestName)
+            $localServiceManifest = ([XML](Get-Content $localServiceManifestPath)).ServiceManifest
+            $diffServicePkgPath = [System.IO.Path]::Combine($diffPackagePath, $localServiceManifest.Name)
```

Comment:
is localServiceManifest.Name same as $serviceManifestImport.ServiceManifestRef.ServiceManifestName 
given that source folder name. If the names need to match; should we use same value in both